pilot/pkg/bootstrap/server.go

```diff
@@ -165,10 +165,13 @@ type Server struct {
 // NewServer creates a new Server instance based on the provided arguments.
 func NewServer(args *PilotArgs) (*Server, error) {
 	e := &model.Environment{
-		ServiceDiscovery: aggregate.NewController(),
-		PushContext:      model.NewPushContext(),
-		DomainSuffix:     args.RegistryOptions.KubeOptions.DomainSuffix,
+		PushContext:  model.NewPushContext(),
+		DomainSuffix: args.RegistryOptions.KubeOptions.DomainSuffix,
 	}
+	ac := aggregate.NewController(&aggregate.Options{
+		MeshFn: e.Mesh,
```

Comment:
Doesn't PushContext already return/contain MeshConfig ? Also I think there is a method to get it - I don't think adding a new mechanism to pass MeshConfig is desirable, we should be consistent.

Not a blocker probably.

Reply:
ack. PushContext does have, but we need it in the aggregate controller